openapi_codec/encode.py

```diff
@@ -115,16 +115,26 @@ def _get_parameters(link, encoding):
                     'required': field.required,
                     'in': 'formData',
                     'description': field.description,
-                    'type': field.type or 'string'
                 }
+                if field.type:
+                    parameter['type'] = field.type
+                    if field.type == 'array':
+                        parameter['items'] = {'type': 'string'}
+                else:
+                    parameter['type'] = 'string'
                 parameters.append(parameter)
             else:
                 # Expand coreapi fields with location='form' into a single swagger
                 # parameter, with a schema containing multiple properties.
                 schema_property = {
                     'description': field.description,
-                    'type': field.type or 'string'
                 }
+                if field.type:
+                    schema_property['type'] = field.type
+                    if field.type == 'array':
+                        schema_property['items'] = {'type': 'string'}
+                else:
+                    schema_property['type'] = 'string'
```

Comment:
Could be neater as...

    ...
        'type': field.type or 'string'  # Leave this as it currently is.
    }
    if field.type == 'array':
        schema_property['items'] = {'type': 'string'}

That'd be a slightly more minimal change.

Otherwise looks good.